### PR TITLE
Refactor tests for SDL2_mixer formats

### PR DESCRIPTION
**Pull request by pudinha**
Parameterize tests so it's easier to extend support to other formats

### CODE REVIEW

**Comment by pudinha** [comment]
Related to #11655

**Comment by pudinha** [comment]
Not sure why the wasm tests don't pass, I don't think I touched anything related... could it be a CI glitch? Those wasm tests refuse to run on my machine, so I couldn't debug.

File: tests/test_browser.py [hunk 1]
```diff
@@ -3200,20 +3200,30 @@ def test_sdl2_misc(self):
...
+  @requires_sound_hardware
+  def test_sdl2_mixer_music_ogg(self):
+    self._test_sdl2_mixer_music_base('ogg', 'alarmvictory_1.ogg')
+
   @requires_sound_hardware
-  def test_sdl2_mixer_mp3(self):
-    shutil.copyfile(path_from_root('tests', 'sounds', 'pudinha.mp3'), 'sound.mp3')
-    self.btest('sdl2_mixer_mp3.c', expected='1', args=['--preload-file', 'sound.mp3', '-s', 'USE_SDL=2', '-s', 'USE_SDL_MIXER=2', '-s', 'SDL2_MIXER_FORMATS=["mp3"]', '-s', 'INITIAL_MEMORY=33554432'])
+  def test_sdl2_mixer_music_mp3(self):
+    self._test_sdl2_mixer_music_base('mp3', 'pudinha.mp3')
```
Comment (kripken) [inline tests/test_browser.py:3226]: these could use `parameterized`, see e.g. `browser.test_gles2_emulation`:
```python
  @parameterized({
    'normal': (['-s', 'FULL_ES2=1'],),
    # Enabling FULL_ES3 also enables ES2 automatically
    'full_es3': (['-s', 'FULL_ES3=1'],)
  })
  def test_gles2_emulation(self, args):
```

Comment (pudinha) [reply tests/test_browser.py]: done! :sparkles:

**Comment by kripken** [comment]
Thanks! Looks good, although when I test locally `interactive.test_sdl2_mixer_music_mp3` hangs on an exception, `Uncaught ReferenceError: reportResultToServer is not defined`.

**Comment by pudinha** [comment]
> Thanks! Looks good, although when I test locally `interactive.test_sdl2_mixer_music_mp3` hangs on an exception, `Uncaught ReferenceError: reportResultToServer is not defined`.

Hmmm this always happened to me, but since CI never complained (are those run on CI?), so I assumed it was something wrong with my setup. Also happens on master, for all formats, even the ones I didn't touch.

```
git checkout master
python tests/runner.py interactive.test_sdl2_mixer
python tests/runner.py interactive.test_sdl2_mixer_mp3
python tests/runner.py interactive.test_sdl2_mixer_wav
```

**Comment by kripken** [comment]
Yes, the problem is that those are not run on CI - a human needs to run the interactive tests (and verify audio actually is played - we don't have a way to automate those atm).

Looks like we haven't run them in a while and they need to be upgraded to use the new `btest` method. E.g. this fixes the first one,
```diff
diff --git a/tests/test_interactive.py b/tests/test_interactive.py
index d2afa0042..1a97f6ea9 100644
--- a/tests/test_interactive.py
+++ b/tests/test_interactive.py
@@ -107,10 +107,7 @@ class interactive(BrowserCore):
 
   def test_sdl2_mixer(self):
     shutil.copyfile(path_from_root('tests', 'sounds', 'alarmvictory_1.ogg'), os.path.join(self.get_dir(), 'sound.ogg'))
-    open(os.path.join(self.get_dir(), 'sdl2_mixer.c'), 'w').write(self.with_report_result(open(path_from_root('tests', 'sdl2_mixer.c')).read()))
-
-    Popen([EMCC, '-O2', '--minify', '0', os.path.join(self.get_dir(), 'sdl2_mixer.c'), '-s', 'USE_SDL=2', '-s', 'USE_SDL_MIXER=2', '-s', 'INITIAL_MEMORY=33554432', '--preload-file', 'sound.ogg', '-o', 'page.html']).communicate()
-    self.run_browser('page.html', '', '/report_result?1')
+    self.btest('sdl2_mixer.c', expected='1', args=['-O2', '-s', 'USE_SDL=2', '-s', 'USE_SDL_MIXER=2', '-s', 'INITIAL_MEMORY=33554432', '--preload-file', 'sound.ogg'])
 
   def test_sdl2_mixer_wav(self):
     shutil.copyfile(path_from_root('tests', 'sounds', 'the_entertainer.wav'), os.path.join(self.get_dir(), 'sound.wav'))
```

I can either open a PR with that, or you can do it in this PR? Up to you, but if I do it it will conflict with this PR which might be more work.

**Comment by pudinha** [comment]
cool! I'll do it now!

**Comment by pudinha** [comment]
@kripken it's done. Just one question though: isn't this the same that's already on test_browser? Maybe we should remove it?

**Comment by kripken** [comment]
> isn't this the same that's already on test_browser? 

There is some duplication, yeah - maybe we can at least remove the code duplication eventually by having it in one place. But we do want to run the tests in browser so that CI tests them continually, and also in interactive so that when we manually  run them we can verify things the test runner can't, like audio/fullscreen/etc.

Maybe we could remove interactive and just tag some tests as being "also interactive" in browser, hmm...

**Pull request merged**